Started roadmap, updated OS request guide for EL9 transition (#735)
CHTC/chtc-website-source: _uw-research-computing/os-transition-htc.md

you are restricted to operating systems used by CHTC, and the number of nodes 
running that operating system. 

### Use Both CentOS Stream 8 (previous default) and CentOS Stream 9 (current default)

To request that your jobs run on computers running **either** version of 
CentOS Linux, add the following requirements line to your submit file:

``` {.sub}
requirements = (OpSysMajorVer == 8) || (OpSysMajorVer == 9)
```

> If you are not comfortable with using class ads in your requirements statement, you can also add one or both of these options to your submit file instead: 

Author: ```suggestion
If you are not comfortable with using class ads in your requirements statement, you can also add **one** of these options to your submit file instead: 
```

```{.sub}

> +WantEL9  # tells HTCondor your job can be matched to CentOS Stream 9 machines

Author: ```suggestion
chtc_want_el9 = true  # tells HTCondor your job should be matched to CentOS Stream 9 machines
# or
```

+WantEL8  # tells HTCondor your job can be matched to CentOS Stream 8 machines
```
> Note: these requirements are not necessary for jobs that use Docker containers; 
> these jobs will run on servers with any operating system automatically. 

The advantage of this option is that you may be able to access a
larger number of computers in CHTC. Note that code compiled on a
newer version of Linux may not run older versions of Linux. Make
sure to test your jobs specifically on both CentOS Stream 8 and CentOS Stream 9
before using the option above.

### Require CentOS 8 (previous default)

To request that your jobs run on servers with CentOS 8 **only** add the
following requirements line to your submit file:

``` {.sub}

> requirements = (OpSysMajorVer == 8)

Author: ```suggestion
requirements = (OpSysMajorVer == 8)
# or 
chtc_want_el8 = true
```

### Use Both CentOS Stream 8 (previous default) and CentOS Stream 9 (current default)

To request that your jobs run on computers running **either** version of 
CentOS Linux, add the following requirements line to your submit file:

``` {.sub}
requirements = (OpSysMajorVer == 8) || (OpSysMajorVer == 9)
```
> Note: these requirements are not necessary for jobs that use Docker containers; 
> these jobs will run on servers with any operating system automatically. 

The advantage of this option is that you may be able to access a
larger number of computers in CHTC. Note that code compiled on a
newer version of Linux may not run older versions of Linux. Make
sure to test your jobs specifically on both CentOS Stream 8 and CentOS Stream 9
before using the option above.

### Require CentOS Stream 8 (previous default) or CentOS Stream 9

To request that your jobs run on servers with CentOS 8 **only**, add one of the
following requirements lines to your submit file:

``` {.sub}
requirements = (OpSysMajorVer == 8)
# or 
chtc_want_el8 = true
```

To request that your jobs run on servers with CentOS 9 **only**, add one of the following requirements lines to your submit file: 

> ```{.sub}

Reviewer: ```suggestion
``` {.sub}
```

requirements = (OpSysMajorVer == 9)
# or

> chtc_want_el9 = true 

Author: ```suggestion
chtc_want_el9 = true 
```

```

You can add the requirements for using CentOS Stream 8 like so:

``` {.submit}
requirements = (Target.HasCHTCStaging == true) && (OpSysMajorVer == 8)
```
Or use the +WantEL syntax: 

``` {.submit}
requirements = (Target.HasCHTCStaging == true)

> +WantEL8 

Author: ```suggestion
chtc_want_el8 = true
```

You can add the requirements for using CentOS Stream 8 like so:

> ``` {.submit}

Reviewer: ```suggestion
``` {.sub}
```

requirements = (Target.HasCHTCStaging == true) && (OpSysMajorVer == 8)
```

> Or use the +WantEL syntax: 

Author: ```suggestion
Or use the `chtc_want_el` syntax: 
```
